File: sort_algorithms/selection_sort/python/selection_sort.py
def selection_sort(list):
    """
    Sorts a list of elements in ascending order using the selection sort algorithm.
    Parameters:
    list (list): The list of elements to be sorted.
    Returns:
    list: The sorted list in ascending order.
    """
    n = len(list);
    for i in range(n):
        min_idx = i;
        for j in range(i + 1, n):
            if list[j] < list[min_idx]:
                min_idx = j
        list[i], list[min_idx] = list[min_idx], list[i];
    
    return list;

not_sort_array = [5, 3, 8, 4, 2];

print('not_sort_array-->', not_sort_array)
print('sort_array-->', selection_sort(not_sort_array))

File: sort_algorithms/selection_sort/javascript/selection_sort.js
const nonSortedArray = [5, 3, 8, 4, 2]
const inputList = [...nonSortedArray]

/**
 * Sorts an array of numbers using the selection sort algorithm.
 *
 * @param {number[]} sortedList - The array of numbers to be sorted.
 * @returns {number[]} The sorted array.
 */

function selection_sort(sortedList) {
    const n = sortedList.length;
    for (let i = 0; i < n - 1; i += 1) {
        let minIdx = i;
        for (let j = i + 1; j < n; j += 1) {
            if (sortedList[j] < sortedList[minIdx]) {
                minIdx = j
            }
        }

        const tempVar = sortedList[i];
        sortedList[i] = sortedList[minIdx];
        sortedList[minIdx] = tempVar;

    }
  return sortedList;
}

console.log('not sorted array: ', nonSortedArray)
console.log('sorted array: ', selection_sort(inputList))


const outputElement = document.getElementById('output');
outputElement.innerHTML = '';
outputElement.innerHTML = `
  <p>Not sorted array: ${nonSortedArray}</p>
  <p>Sorted array: ${inputList}</p>
  <p>For details, check the console.</p>
`;